astropy/units/physical.py

```diff
@@ -1,133 +1,473 @@
+    included in one string if they are separated with a slash (e.g.,
+    ``"momentum/impulse"``).  String representations of physical types
+    may include underscores instead of spaces.
+
+    Examples
+    --------
+    The preferred method to access a physical type is by accessing the
+    ``physical_type`` attribute of a unit.
+
+    >>> import astropy.units as u
+    >>> u.meter.physical_type
+    PhysicalType('length')
+
+    Some units are dimensionally compatible with multiple physical types.
+    A pascal is intended to represent pressure or stress, but the unit
+    decomposition is equivalent to that of energy density.
+
+    >>> pressure = u.Pa.physical_type
+    >>> pressure
+    PhysicalType({'energy density', 'pressure', 'stress'})
+    >>> 'energy density' in pressure
+    True
+
+    Physical types can be tested for equality against other physical
+    type objects or against strings that may contain the name of a
+    physical type.
+
+    >>> area = (u.m ** 2).physical_type
+    >>> area == u.barn.physical_type
+    True
+    >>> area == "area"
+    True
+
+    Multiplication, division, and exponentiation are enabled so that
+    physical types may be used for dimensional analysis.
+
+    >>> length = u.pc.physical_type
+    >>> area = (u.cm ** 2).physical_type
+    >>> length * area
+    PhysicalType('volume')
+    >>> area / length
+    PhysicalType('length')
+    >>> length ** 3
+    PhysicalType('volume')
+
+    Unknown physical types are labelled as ``"unknown"``.
+
+    >>> (u.s ** 13).physical_type
+    PhysicalType('unknown')
+
+    Dimensional analysis may be performed for unknown physical types too.
+
+    >>> length_to_19th_power = (u.m ** 19).physical_type
+    >>> length_to_20th_power = (u.m ** 20).physical_type
+    >>> length_to_20th_power / length_to_19th_power
+    PhysicalType('length')
+    """
+
+    def __init__(self, unit, physical_types):
+        self._unit = _replace_temperatures_with_kelvin(unit)
+        self._physical_type_id = self._unit._get_physical_type_id()
+        self._physical_type = _standardize_physical_type_names(physical_types)
+        self._physical_type_list = sorted(self._physical_type)
+
+    def __iter__(self):
+        yield from self._physical_type_list
+
+    def __eq__(self, other):
+        """
+        Return `True` if ``other`` represents a physical type that is
+        consistent with the physical type of the `PhysicalType` instance.
+        """
+        if isinstance(other, PhysicalType):
+            return self._physical_type_id == other._physical_type_id
+        elif isinstance(other, str):
+            other = _standardize_physical_type_names(other)
+            return other.issubset(self._physical_type)
+        else:
+            return NotImplemented
+
+    def __ne__(self, other):
+        equality = self.__eq__(other)
+        return not equality if isinstance(equality, bool) else NotImplemented
+
+    @property
+    def _name_string_as_ordered_set(self):
```

Comment:
Absolute nitpick that you should feel free to ignore, but I'd keep this a method rather than a property (just to avoid the indirection).